packages/middleware/src/lib/Middleware.ts

```diff
@@ -228,17 +221,13 @@ export class Middleware<ContextData extends NonNullObject, StoredValue = unknown
     return payload;
   }
 
-  public run<P extends Payload>(payload: P): Awaitable<unknown> {
```

Comment:
Why remove run?

Reply:
I don't see any real purpose for it since it doesn't modify the payload. The only thing that might make sense for this method is logging, but even then you still need to do a bunch of checks on what the payload is to get any of the properties.

Reply:
Right but for the case of logging now, it would have to have custom logic for each method to then run a simple log which seems overcomplicated, perhaps a simple run command that provides a simple payload for all methods?

Reply:
All payloads extend `Payload`, so whatever is from that is what you get guaranteed. I don't think it would be good for end-users to need to check the payload rather then to use direct methods for each payload type.

Reply:
I agree that direct methods is better, but something like @PreEverything() would be useful

Reply:
Can you elaborate in Discord? If this is a whole new method I think a separate PR is in order for this.